### DISCUSSION TITLE
Help with uploading image file into pocketbase collection

### DISCUSSION THREAD

**Discussion by rameshkumar70** [pocketbase/pocketbase#4841] [Q&A] [answered]
Hi,

I am using the following code at the client (sveltekit's +page.svelte) to send a form to the server:
`const formData = new FormData(evt.currentTarget);`
`formData.set('profilePicture', picUploadInput.files[0], 'myfile.png');`
`const response = await fetch(``/students/${dbRecord.id}/personal/``, {`
	`method: 'POST',`
	`body: formData`
`});`

At the sveltekit server/backend, I am able to receive the formdata using the following code:
`const form = await request.formData();`
`const editedStudentData = {`
        `name: form.get('name'),`
        `email: form.get('email'),`
        `mobile: form.get('mobile'),`
        `gender: form.get('gender'),`
        `profilePicture: form.get('profilePicture')`
`};`
`for (let pair of form.entries()) {`
	`console.log(pair[0] + ', ' + pair[1]);`
`}`

The `console.log()` in the loop of the above code prints out the following output, and as can be seen, the `profilePicture` field does contain a file object which I am trying to upload into Pocketbase: 

`id, v2yxx2q60izooey`
`name, Chin`
`gender, M`
`mobile, 27892378u9`
`email, chin@mail.com`
`profilePicture, [object File]`

The problem I'm facing is when I invoke an update of the record in the pocketbase collection, the file I am trying to upload is not getting stored/updated in the collection. The below code is what I am using to update the record:

`await locals.userPb.collection('Student').update(studentId, editedStudentData);`

And the following is the debug trace of the pocketbase server during the update:

```
INFO PATCH /api/collections/Student/records/v2yxx2q60izooey
[2.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = '_pb_users_auth_' OR LOWER([[name]])='_pb_users_auth_' LIMIT 1
[0.00ms] SELECT `users`.* FROM `users` WHERE `users`.`id`='193rqht10ptk6zm' LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = '7vfpthon2jsteho' OR LOWER([[name]])='7vfpthon2jsteho' LIMIT 1
[0.00ms] SELECT DISTINCT `Student`.* FROM `Student` LEFT JOIN `users` `__auth_users` ON `__auth_users`.`id`='193rqht10ptk6zm' WHERE (`Student`.`id`='v2yxx2q60izooey') AND ([[__auth_users.active]] = 1) LIMIT 1
[12.00ms] UPDATE `Student` SET `active`=true, `created`='2024-02-27 03:18:26.671Z', `email`='chin@mail.com', `gender`='M', `id`='v2yxx2q60izooey', `mobile`='27892378u9', `name`='Chin', `profilePicture`='', `updated`='2024-04-28 15:47:23.685Z' WHERE `id`='v2yxx2q60izooey'
```

I'm out of ideas on what to try next and I would appreciate any pointers/help with this.

**Comment by ganigeorgiev** [comment]
I'm not sure what could be the cause of the issue in your case.

Note that support for file upload using plain object notation was added in JS SDK v0.17.0, so double check that you use the latest JS SDK and PocketBase versions.

Additionally you don't have to copy the FormData fields into a plain object in order to submit them and instead you can simply send the form directly since its field names match with the ones from the collection:

```js
await locals.userPb.collection('Student').update(studentId, form);
```

**Reply by rameshkumar70** [reply]
Thank you for the lightning fast reply. 
The version of pocketbase I am using is the latest: 
`$ ./pocketbase -v`
`pocketbase version 0.22.10`

But how do I check the JS SDK version?

**Reply by ganigeorgiev** [reply, marked as answer]
Check in your `package-lock.json` and search for `node_modules/pocketbase` entry and/or run `npm update` to update your dependencies.

**Reply by ganigeorgiev** [reply]
Alternatively, if you are using the latest JS SDK and the object notation still doesn't work (_for example because of an older Node version and FormData polyfill that doesn't expose the `Blob/File` globally_) you can try to create your own `FormData` instance and submit that object.

```js
const editedStudentData = new FormData();
// pick the fields you want to update
editedStudentData.append("name", form.get("name"))
...
```

**Reply by rameshkumar70** [reply]
Yes! It was the JS SDK version that was the culprit.....I've been using 0.16.0. It never dawned on me to update the SDK whenever I updated the pocketbase server. It's working perfectly now, with my SDK updated  to 0.21.2.
Thank you very much for being so responsive and helpful.
